src/compiler/transformers/es6.ts

```diff
@@ -357,7 +357,13 @@ namespace ts {
                     return visitParenthesizedExpression(<ParenthesizedExpression>node, /*needsDestructuringValue*/ true);
 
                 case SyntaxKind.BinaryExpression:
-                    return visitBinaryExpression(<BinaryExpression>node, /*needsDestructuringValue*/ true);
+                    // We should only add parenthesis around expression when the node is within a binaryExpression.

```

Comment:
This would better be handled in one of the `parenthesize*` functions in `factory.ts`, which is where all of the parenthesization rules take place. I'm trying to eventually get rid of the need for `currentParent`.